packages/utils/interactive-hooks/src/useAsToggleWithEvent.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { AccessibilityActionEvent, GestureResponderEvent } from 'react-native';
+import { KeyPressEvent } from '.';
+
+export type OnToggleWithEventCallback = (e: GestureResponderEvent | KeyPressEvent | AccessibilityActionEvent, value?: boolean) => void;
+export type OnChangeWithEventCallback = (e: GestureResponderEvent | KeyPressEvent | AccessibilityActionEvent) => void;
+
+/* Re-usable hook for toggle components.
```

Comment:
I made a copy of this to be able to leave the version used in the old checkbox alone. If you have suggestions on how to handle this better, let me know.

Reply:
Maybe once the old one is deprecated, we could rename this one to just `useAsToggle` ? `WithEvent` feels a bit unnecessary. I'm not sure my suggestion isn't a breaking change though.

Reply:
I think we do technically ship this package so it could be breaking? I don't think it's used a lot outside of our repo though. Maybe it's fine as long as we bump the major version